### DISCUSSION TITLE
script kiddie stumbling through obsidian.nvim with a templates question.

### DISCUSSION THREAD

**Discussion by gchingon** [epwalsh/obsidian.nvim#565] [Q&A]
I'm just a prose writer on Mac OS, that's dumb enough to like neovim over GUI editors. 
I've got obsidian.nvim working, other than templates. What I'm trying to do is have my templates in my .config directory, which I back up on git. My issue is trying to trigger templates, I get an error `Templates folder is not defined or does not exist` folder DOES exist and there are various template.md files in said folder. I imagine I don't have the `templates` variable in the right place, or `folder` variable doesn't work / isn't recognized and only `subdir` is allowed. if the latter is the case, then I guess this is a feature request. if I can use `folder` and define templates dir location to be anywhere I wish how can I fix this?

```lua
return {
  'epwalsh/obsidian.nvim',
  version = '*', -- recommended, use latest release instead of latest commit
  lazy = true,
  ft = 'markdown',
  dependencies = {
    'nvim-lua/plenary.nvim',
  },
  opts = {
    workspaces = {
      {
        name = 'notes',
        path = '$HOME/Documents/notes',
        daily_notes = {
          folder = '%Y/%m/',
        },
      },
      {
        name = 'blog',
        path = '$HOME/Documents/blog/',
      },
      {
        name = 'novel',
        path = '$HOME/Documents/novel/',
        'prepend_new_note_path',
      },
    },
  },

  ui = {
    enable = true,
    update_debounce = 500,
  },

  completion = {
    nvim_cmp = true,
    min_chars = 3,
  },

  keys = {
    { '<leader>on', '<cmd>ObsidianNew<cr>', desc = 'New Obsidian note', mode = 'n' },
    { '<leader>oo', '<cmd>ObsidianSearch<cr>', desc = 'Search Obsidian notes', mode = 'n' },
    { '<leader>ob', '<cmd>ObsidianBacklinks<cr>', desc = 'Show location list of backlinks', mode = 'n' },
    { '<leader>ot', '<cmd>ObsidianTemplate<cr>', desc = 'Follow link under cursor', mode = 'n' },
  },

  templates = {
    folder = '$HOME/.config/templates',
    date_format = '%Y-%m-%d',
    time_format = '%H:%M',
  },
}
```

**Comment by epwalsh** [comment]
Hey @gallo-s-chingon, previously the templates folder was assumed to be relative to your vault root. However as of dbc2b0a this is now supported. If you pull the latest commit it should work for you. If not, let me know.

**Reply by gchingon** [reply]
thanks for the reply, I'll update once I'm home and report back

**Reply by gchingon** [reply]
@epwalsh I got obsidian.nvim updated to `db41b1f` chore (release) v3.7.12

still get the `Templates folder is not defined or does not exist` even changed $HOME to `/Users/username` just in case that could be the cause. so I'm guessing I have templates indented wrong / in the wrong place.

**Reply by epwalsh** [reply]
Can you post the output of `:ObsidianDebug`?

**Reply by gchingon** [reply]
here's the output of `:ObsidianDebug`

```
Error executing Lua callback: ...ocal/share/nvim/lazy/obsidian.nvim/lua/obsidian/path.lua:402: FileNotFoundError: /Users/gch
ingon/Documents/podcast
stack traceback:
        [C]: in function 'error'
        ...ocal/share/nvim/lazy/obsidian.nvim/lua/obsidian/path.lua:402: in function 'resolve'
        ...share/nvim/lazy/obsidian.nvim/lua/obsidian/workspace.lua:79: in function 'new_from_spec'
        .../nvim/lazy/obsidian.nvim/lua/obsidian/commands/debug.lua:47: in function 'func'
        ...e/nvim/lazy/obsidian.nvim/lua/obsidian/commands/init.lua:68: in function <...e/nvim/lazy/obsidian.nvim/lua/obsidi
an/commands/init.lua:67>
```
